model/ChangePermissionsService.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2023 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoDacSimple\model;
+
+use core_kernel_classes_Resource;
+use oat\generis\model\OntologyRdfs;
+use oat\oatbox\event\EventManager;
+use oat\tao\model\event\DataAccessControlChangedEvent;
+use oat\taoDacSimple\model\event\DacAffectedUsersEvent;
+use oat\taoDacSimple\model\event\DacRootChangedEvent;
+
+class ChangePermissionsService
+{
+    private DataBaseAccess $dataBaseAccess;
+    private PermissionsStrategyInterface $strategy;
+    private EventManager $eventManager;
+
+    public function __construct(
+        DataBaseAccess $dataBaseAccess,
+        PermissionsStrategyInterface $strategy,
+        EventManager $eventManager
+    ) {
+        $this->dataBaseAccess = $dataBaseAccess;
+        $this->strategy = $strategy;
+        $this->eventManager = $eventManager;
+    }
+
+    public function change(core_kernel_classes_Resource $resource, array $permissionsToSet, bool $isRecursive): void

```

Comment:
- [ ] Suggestion: To be extensible, can we use a CommandObject instead? I think @hectoras has already an object for that created.

Reply:
Command added